standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -2419,9 +2419,9 @@ private void create_table_core(final RawStore ms, final CreateTableRequest req)
       }
 
       if (!TableType.VIRTUAL_VIEW.toString().equals(tbl.getTableType())) {
-        if (tbl.getSd().getLocation() == null
-            || tbl.getSd().getLocation().isEmpty()) {
-          tblPath = wh.getDefaultTablePath(db, tbl);
+        if (tbl.getSd().getLocation() == null || tbl.getSd().getLocation().isEmpty()) {
+          String relPath = tbl.getTableName() + (tbl.isSetTxnId() ? "_v" + tbl.getTxnId() : "");
+          tblPath = wh.getDefaultTablePath(db, relPath, isExternal(tbl));
```

Comment:
can't we have a variation of this method which accepts a full "table" - and can decide the path ; and the externalability based on that...so that we don't spread the above line everywhere this method is invoked (I'm also working on a patch which invokes this method)

Reply:
moved this logic to the Warehouse.getDefaultTablePath